src/TwitchLib.Communication/Clients/AClientBase.cs

```diff
@@ -27,10 +27,11 @@ namespace TwitchLib.Communication.Clients
     ///         </item>
     ///     </list>
     /// </summary>
-    public abstract class AClientBase : IClient
+    public abstract class AClientBase<T> : IClient where T : IDisposable
```

Comment:
Not sure if I missed a recent convention but can we rename that to just ClientBase or sth and drop the A?